File: steps/python/streams_adoptions_stream.py
# steps/python/streams_adoptions_stream.step.py
config = {
    "name": "adoptions",
    "schema": {
        "entityId": "string",
        "type": ["application", "pet"],
        "phase": "string",
        "message": "string?"
    },
    "baseConfig": { "storageType": "default" }
}

File: steps/typescript/streams.adoptions.stream.ts
import { StreamConfig } from 'motia';
import { z } from 'zod';

export const config: StreamConfig = {
  name: 'adoptions',
  schema: z.object({
    entityId: z.string(),
    type: z.enum(['application', 'pet']),
    phase: z.string(),
    message: z.string().optional(),
  }),
  baseConfig: { storageType: 'default' },
};
